internal/controller/account_resolver.go

```diff
@@ -98,7 +108,7 @@ func (bar *BOPAccountIdResolver) MapClientIdToAccountId(ctx context.Context, cli
 			logger.Debugf("Error Response: %p", errResponse)
 			return "", "", "", fmt.Errorf("Unable to find account: %w", err)
 		}
-		return "", "", "", err
+		return "", "", "", fmt.Errorf("Unable to find account: %s", errResponse)
```

Comment:
If we got into this code block, then we know that the call failed and we need to return an error.